### DISCUSSION TITLE
Forum and User data differences for v1.0.4 to 2.3.3 upgrade

### DISCUSSION THREAD

**Discussion by travbowls** [e107inc/e107#5341] [Q&A Updating] [answered]
I'm new to this group, but wanted to share a few things I've found when I tried upgrading from v1.0.4 to 2.3.3.   I tried the upgrade zip and just replacing with the newer files, but my application had many old plugins and theme's that didn't seem supported anymore.  I'm no PHP export, so I decided to tackle this from the data side.  So I wiped everything except images and few things, upgraded PHP to 8.1, created a new database with MySQL8.0 and just redid some of the basic e107 configuration.  Then I exported certain info from the tables I needed from the old database, and imported or inserted into the new database.  This was nice to have a preserved backup of the DB I can reference at any time.  I will try to add more details in subsequent posts.  Not sure if anyone is interested, but figured I'd throw this out there.

These are are the high level details:
-e107_forum_t was the old thread table.  Threads with no Parent were a top level thread.  Threads with a parent where child threads (posts).  This became e107_forum_thread and e107_forum_post in the new tables where thread is the parent, post is the child records.   
-User tables (e107_user, e107_user_class_classes, etc) had a few different columns, extended classes differences, etc.

The AjaxThanks plugin didn't work anymore, so my next goal is to try to fix that plugin to work with e107 2.X, and port over all the old data from that, for people to Thank/Like posts.

**Comment by travbowls** [comment]
**_THREADS_**

![forumThreadTableDifferences](https://github.com/user-attachments/assets/fb645aec-2482-460d-a98c-cc0f0f2cb779)

I exported the whole e107_forum_t table and imported it into the new database, so I could use standard SQL to move files from it to the thread and posts tables.  Then performed the following SQLs

------old Threads to new Threads--------
INSERT INTO `e107_forum_thread`(`thread_id`, `thread_name`, `thread_forum_id`, `thread_views`, `thread_active`, `thread_lastpost`, `thread_sticky`, `thread_datestamp`, `thread_user`, `thread_lastuser`, `thread_total_replies`) 
SELECT `thread_id`, `thread_name`, `thread_forum_id`, `thread_views`, `thread_active`, `thread_lastpost`, `thread_s`, `thread_datestamp`, `thread_user`, `thread_lastuser`, `thread_total_replies`
FROM e107_forum_t
WHERE thread_parent =0;

------old Threads to new Posts-----------
INSERT INTO `e107_forum_post`
(`post_id`, `post_entry`, `post_thread`, 
`post_forum`, `post_datestamp`, 
`post_user`, `post_edit_datestamp`, `post_edit_user`) 
SELECT 
`thread_id`, `thread_thread`,`thread_parent`, 
`thread_forum_id`, `thread_datestamp`,
`thread_user`, `thread_edit_datestamp`, `thread_lastuser`
FROM `e107_forum_t` WHERE thread_parent != 0;


Then I realized, the parent threads also needed to be copied over as a post, so you can see the posters initial post entry when they started the thread.  Note that the post_id and post_thread are the same value for the root level posts of a thread.
------old PARENT Threads to new Posts-----------
INSERT INTO `e107_forum_post`
(`post_id`, `post_entry`, `post_thread`, 
`post_forum`, `post_datestamp`, 
`post_user`, `post_edit_datestamp`, `post_edit_user`) 
SELECT 
`thread_id`, `thread_thread`,`thread_id`, 
`thread_forum_id`, `thread_datestamp`,
`thread_user`, `thread_edit_datestamp`, `thread_lastuser`
FROM `e107_forum_t` WHERE thread_parent = 0;

Backup your database/tables before you try any of these, I can't guarantee it's perfect, but for me, it copied over all the data I needed.

**Reply by tgtje** [reply]
@travbowls nice job and informative.
Personally i do it somewhat different, but in base it is similar.

edit: the practice will be similar for other parts also ..just look out for correct insertion names

**Reply by travbowls** [reply]
Edit:  Sorry, I thought there was an issue with the parent forum nesting, but just realized there's an issue with the breadcrumbs when 3 levels in, so it was tricking me.

**Comment by Moc** [comment, marked as answer]
Hi there and welcome :) 

There is an extensive forum upgrade routine that must be followed when upgrading from v1 to v2. The basic procedure is as follows: 
- Create backup of files and database
- Uninstall and delete any plugins that are not in use
- Make sure you are using at least PHP 7.4 (but preferably PHP 8). 
- Overwrite the existing installation with upgrade files (https://e107.org/download) 
- Go to Admin Area, and you'll be prompted with instructions such as database updates. 

The database (and other checks and changes) can be found in this file: https://github.com/e107inc/e107/blob/master/e107_plugins/forum/forum_update.php

Making manual changes to the database is of course your own choice, but please note that any issues resulting from your changes are not easily fixed and support given may be limited. I would strongly recommend using the upgrade methods provided by the upgrade/installation package. In your case, at the very least, you should check the forum_upgrade.php file to check for all other changes that must be made. 

Also please use the database tools provided in Tools > Database to check for integrity etc.
